pyanaconda/modules/common/structures/partitioning.py

```diff
@@ -18,10 +18,232 @@
+
+
+class PartitioningRequest(DBusData):
+    """Partitioning request data."""
+
+    def __init__(self):
+        self._partitioning_scheme = DEFAULT_AUTOPART_TYPE
+        self._file_system_type = ""
+        self._excluded_mount_points = []
+
+        self._encrypted = False
+        self._passphrase = ""
+        self._cipher = ""
+        self._luks_version = ""
+
+        self._pbkdf = ""
+        self._pbkdf_memory = 0
+        self._pbkdf_time = 0
+        self._pbkdf_iterations = 0
+
+        self._escrow_certificate = ""
+        self._backup_passphrase_enabled = False
+
+    @property
+    def partitioning_scheme(self) -> Int:
+        """The partitioning scheme.
+
+        Allowed values:
+            0  Create regular partitions.
+            1  Use the btrfs scheme.
+            2  Use the lvm scheme.
+            3  Use the lvm thinp scheme.
+
+        :return: an number of the partitioning scheme
+        """
+        return self._partitioning_scheme
+
+    @partitioning_scheme.setter
+    def partitioning_scheme(self, scheme: Int):
+        self._partitioning_scheme = scheme
+
+    @property
+    def file_system_type(self) -> Str:
+        """Type of a file system used on the partitions.
+
+        For example: ext4
+
+        :return: a name of a file system type
+        """
+        return self._file_system_type
+
+    @file_system_type.setter
+    def file_system_type(self, value: Str):
+        self._file_system_type = value
+
+    @property
+    def excluded_mount_points(self) -> List[Str]:
+        """Excluded mount points.
+
+        Don't create partitions for the excluded
+        mount points during the partitioning.
+
+        For example: /home, /boot, swap
+
+        :return: a list of mount points
+        """
+        return self._excluded_mount_points
+
+    @excluded_mount_points.setter
+    def excluded_mount_points(self, mount_points: List[Str]):
+        self._excluded_mount_points = mount_points
+
+    @property
+    def encrypted(self) -> Bool:
+        """Should devices be encrypted?
+
+        :return: True or False
+        """
+        return self._encrypted
+
+    @encrypted.setter
+    def encrypted(self, encrypted: Bool):
+        self._encrypted = encrypted
+
+    @property
+    def passphrase(self) -> Str:
+        """Passphrase for all encrypted devices.
+
+        :return: a string with the passphrase
+        """
+        return self._passphrase
+
+    @passphrase.setter
+    def passphrase(self, value: Str):
+        self._passphrase = value
+
+    @property
+    def cipher(self) -> Str:
+        """Encryption algorithm used to encrypt the filesystem.
+
+        For example: aes-xts-plain64
+
+        :return: a name of an algorithm
+        """
+        return self._cipher
+
+    @cipher.setter
+    def cipher(self, cipher: Str):
+        self._cipher = cipher
+
+    @property
+    def luks_version(self) -> Str:
+        """Version of LUKS.
```

Comment:
do we have something else than luks1 & luks2 ? I guess list of valid values could be specified.

Reply:
Fixed.